go/chat/boxer.go

```diff
@@ -227,7 +240,7 @@ func (b *Boxer) UnboxMessage(ctx context.Context, boxed chat1.MessageBoxed, conv
 	keyMembersType := b.getEffectiveMembersType(ctx, boxed, conv.GetMembersType())
 	nameInfo, err := CtxKeyFinder(ctx, b.G()).FindForDecryption(ctx,
 		tlfName, boxed.ClientHeader.Conv.Tlfid, conv.GetMembersType(),
-		boxed.ClientHeader.TlfPublic, boxed.KeyGeneration, keyMembersType == chat1.ConversationMembersType_KBFS)
+		conv.IsPublic(), boxed.KeyGeneration, keyMembersType == chat1.ConversationMembersType_KBFS)
```

Comment:
Does this come from a signed message? I just want to make sure the server can't convince you to encrypt a message incorrectly by claiming a private conv is public

Reply:
Ok changed it, now it just gives a permanent error back for the message instead of causing a failure in `FindForDecryption`